test/sql-tap/array.test.lua

```diff
@@ -1035,21 +1035,6 @@ test:do_test(
         {1, 2, 3}
     })
 
-local remote = require('net.box')
-box.cfg{listen = os.getenv('LISTEN')}
-box.schema.user.grant('guest', 'execute', 'sql')
-local cn = remote.connect(box.cfg.listen)
-test:do_test(
-    "builtins-14.2",
```

Comment:
I would move `builtins-14.1` (it is just above) as well, because these test cases are about the same topic: the only difference that one targets a local box.execute, while another targets iproto execute. *Feel free to ignore.*

Reply:
Agree, fixed:

```diff
diff --git a/test/sql-luatest/array_test.lua b/test/sql-luatest/array_test.lua
index 2fd13cf83b..fe40e3d12b 100644
--- a/test/sql-luatest/array_test.lua
+++ b/test/sql-luatest/array_test.lua
@@ -12,6 +12,15 @@ g.after_all(function()
     g.server:stop()
 end)
 
+-- Make sure that ARRAY values can be used as bound variable.
+g.test_builtins_14_1 = function()
+    g.server:exec(function()
+        local sql = [[SELECT #a;]]
+        local arg = {{['#a'] = {1, 2, 3}}}
+        t.assert_equals(box.execute(sql, arg).rows[1][1], {1, 2, 3})
+    end)
+end
+
 g.test_builtins_14_2 = function()
     local conn = g.server.net_box
     local ok, res = pcall(conn.execute, conn, [[SELECT #a;]],
diff --git a/test/sql-tap/array.test.lua b/test/sql-tap/array.test.lua
index 042c8420ee..2831f31c01 100755
--- a/test/sql-tap/array.test.lua
+++ b/test/sql-tap/array.test.lua
@@ -1025,16 +1025,6 @@ test:do_execsql_test(
         "array"
     })
 
--- Make sure that ARRAY values can be used as bound variable.
-test:do_test(
-    "builtins-14.1",
-    function()
-        local res = box.execute([[SELECT #a;]], {{['#a'] = {1, 2, 3}}})
-        return {res.rows[1][1]}
-    end, {
-        {1, 2, 3}
-    })
-
 box.execute([[DROP TABLE t1;]])
 box.execute([[DROP TABLE t;]])
 
```